vorbis_codec_registration.src.html

```diff
@@ -0,0 +1,98 @@
+    intend to implement or use a specific codec format. Implementers of
+    WebCodecs are not required to support the Vorbis codec.
+
+    This registration is non-normative.
+
+Markup Shorthands:css no, markdown yes, dfn yes
+!Participate: <a href="https://github.com/w3c/webcodecs">Git Repository.</a>
+!Participate: <a href="https://github.com/w3c/webcodecs/issues/new">File an issue.</a>
+!Version History: <a href="https://github.com/w3c/webcodecs/commits">https://github.com/w3c/webcodecs/commits</a>
+</pre>
+
+<pre class='anchors'>
+spec: WEBCODECS; urlPrefix: https://w3c.github.io/webcodecs/#
+    type: attribute
+        for: EncodedAudioChunkType; text: key; url: dom-encodedaudiochunktype-key
+        text: AudioDecoderConfig.description; url: dom-audiodecoderconfig-description
+        text: AudioDecoderConfig.sampleRate; url: dom-audiodecoderconfig-samplerate
+        text: AudioDecoderConfig.channelCount; url: dom-audiodecoderconfig-channelcount
+    type: dfn
+        for: EncodedAudioChunk; text: [[internal data]]; url: dom-encodedaudiochunk-internal-data-slot
+        for: EncodedAudioChunk; text: [[type]]; url: dom-encodedaudiochunk-type-slot
+    type: interface
+        text: EncodedAudioChunk; url: encodedaudiochunk
+    type: dictionary
+        text: AudioDecoderConfig; url: dictdef-audiodecoderconfig
+</pre>
+
+<pre class='biblio'>
+{
+  "VORBIS": {
+    "href": "https://xiph.org/vorbis/doc/Vorbis_I_spec.html",
+    "title": "Vorbis I specification",
+    "publisher": "Xiph.Org Foundation",
+    "date": "July 4, 2020"
+  }
+}
+</pre>
+
+Fully qualified codec strings {#fully-qualified-codec-strings}
+==============================================================
+
+The codec string is `"vorbis"`.
+
+EncodedAudioChunk data {#encodedaudiochunk-data}
+================================================
+
+{{EncodedAudioChunk}} [=EncodedAudioChunk/[[internal data]]=] is expected to be
+an "audio packet", as described in the section 4.3 of the [[VORBIS]] specification.
+
+AudioDecoderConfig description {#audiodecoderconfig-description}
+================================================================
+
+{{AudioDecoderConfig.description}} is required. It is assumed to be three
+Vorbis header packets, respectively the identification header, the comments
+header, and the setup header, in this order, as described in section 4.2 of [[VORBIS]].
+
+The {{AudioDecoderConfig.sampleRate}} and {{AudioDecoderConfig.channelCount}}
+members are overridden by what the decoder finds in the identification header.
+
+NOTE: The comments header content is not used by [[WEBCODECS]].
+
+EncodedAudioChunk type {#encodedaudiochunk-type}
+================================================
+
+The [=EncodedAudioChunk/[[type]]=] is not used in this codec, its value is
+ignored.
+
+NOTE: Once the initialization has succeeded, any Vorbis packet can be decoded at
+any time, not necessarily in sequence, there is no concept of key-frames.
```

Comment:
In my mind, this is the same as saying that all Vorbis chunks should have type `key`.

Reply:
Yes, I went back and forth between "every packet is a key" or "this doesn't matter at all". I decided to simply ignore it, but I could certainly be convinced of the opposite behaviour.